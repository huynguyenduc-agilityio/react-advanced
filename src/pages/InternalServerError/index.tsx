import { Link } from 'react-router-dom';
import { Box, Button, Center, Heading, Text } from '@chakra-ui/react';

// Constants
import { PUBLIC_ROUTERS } from '@/constants';

const InternalServerError = () => {
  return (
    <Center py="300px">
      <Box textAlign="center">
        <Heading as="h1" color="secondary.400" fontSize="2xl">
          500 Internal Server Error
        </Heading>
        <Text
          color="secondary.300"
          fontSize="lg"
          fontWeight="medium"
          maxW="414px"
          mt="10px"
        >
          Sorry, something went wrong. :(
        </Text>

        <Button variant="primary" mt="42px" as={Link} to={PUBLIC_ROUTERS.ROOT}>
          Back to Dashboard
        </Button>
      </Box>
    </Center>
  );
};

export default InternalServerError;
