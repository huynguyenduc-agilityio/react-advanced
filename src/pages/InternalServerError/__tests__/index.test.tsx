import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

// Pages
import InternalServerError from '..';

describe('InternalServerError Page', () => {
  test('should render correctly', () => {
    const view = render(
      <BrowserRouter>
        <InternalServerError />
      </BrowserRouter>,
    );

    expect(view).toMatchSnapshot();
  });
});
