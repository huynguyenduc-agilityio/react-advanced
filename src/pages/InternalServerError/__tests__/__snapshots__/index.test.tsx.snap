// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalServerError Page should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1j38vg2"
      >
        <div
          class="css-xi606m"
        >
          <h1
            class="chakra-heading css-1am5goc"
          >
            500 Internal Server Error
          </h1>
          <p
            class="chakra-text css-1fu8yun"
          >
            Sorry, something went wrong. :(
          </p>
          <a
            class="chakra-button css-xqsa59"
            href="/"
          >
            Back to Dashboard
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1j38vg2"
    >
      <div
        class="css-xi606m"
      >
        <h1
          class="chakra-heading css-1am5goc"
        >
          500 Internal Server Error
        </h1>
        <p
          class="chakra-text css-1fu8yun"
        >
          Sorry, something went wrong. :(
        </p>
        <a
          class="chakra-button css-xqsa59"
          href="/"
        >
          Back to Dashboard
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
